docs/sql-ref-datatypes.md

```diff
@@ -49,6 +49,37 @@ Spark SQL and DataFrames support the following data types:
   absolute point in time.
   - `DateType`: Represents values comprising values of fields year, month and day, without a
   time-zone.
+* Interval types
+  - `YearMonthIntervalType(startField, endField)`: Represents a year-month interval which is made up of a contiguous subset of the following fields:
+    - MONTH, months within years `[0..11]`,
+    - YEAR, years in the range `[0..178956970]`.
+
+    Individual interval fields are non-negative, but an interval itself can have a sign, and be negative.
+
+    `startField` is the leftmost field, and `endField` is the rightmost field of the type. Valid values of `startField` and `endField` are 0(MONTH) and 1(YEAR). Supported year-month interval types are:
+      - `YearMonthIntervalType(YEAR, YEAR)` or `YearMonthIntervalType(YEAR)`. For instance, a value of the type is `INTERVAL '2021' YEAR`.
+      - `YearMonthIntervalType(YEAR, MONTH)`. For instance, a value of the type is `INTERVAL '2021-07' YEAR TO MONTH`.
+      - `YearMonthIntervalType(MONTH, MONTH)` or `YearMonthIntervalType(MONTH)`. For example, `INTERVAL '10' MONTH` is a value of the type.
```

Comment:
Use same `For example` or `For instance`?

Reply:
I think it would be better to convert the list to a table with table title.

Reply:
> I think it would be better to convert the list to a table with table title.

Looks better.